File: app/src/main/java/com/example/photogallery_lab10/PhotoApi.java
package com.example.photogallery_lab10;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

interface PhotoApi {

    @GET("services/rest/")
    Call<FlickrResponse> getInterestingPhotos(
            @Query("method") String method,
            @Query("api_key") String apiKey,
            @Query("format") String format,
            @Query("nojsoncallback") int noJsonCallback
    );
}

File: app/src/main/java/com/example/photogallery_lab10/PhotoApi.kt
package com.example.photogallery_lab10;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

interface PhotoApi {
    @GET("?")
    fun getInterestingPhotos(
            @Query("method") method: String = "flickr.interestingness.getList",
            @Query("api_key") apiKey: String,
            @Query("format") format: String = "json",
            @Query("nojsoncallback") noJsonCallback: Int = 1,
            @Query("per_page") perPage: Int = 100,
            @Query("page") page: Int = 1,
            @Query("extras") extras: String = "url_s"
    ): Call<FlickrResponse>
}
